### PR TITLE
FQF remove methods from EG node

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/se/checks/ConditionAlwaysTrueOrFalseCheck.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public void checkEndOfExecution(CheckerContext context) {
 
   private static Set<List<JavaFileScannerContext.Location>> flowFromNode(ExplodedGraph.Node node, boolean conditionIsAlwaysTrue) {
     List<Class<? extends Constraint>> domains = Lists.newArrayList(ObjectConstraint.class, BooleanConstraint.class);
-    Set<List<JavaFileScannerContext.Location>> flows = FlowComputation.flow(node.parent(), node.programState.peekValue(), domains);
+    Set<List<JavaFileScannerContext.Location>> flows = FlowComputation.flow(node, node.programState.peekValue(), domains);
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/ConditionAlwaysTrueOrFalseCheck.java]: This modification looks a bit suspicious without a testcase and definitely does not sound like a quality flaw fix.

File: java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java [hunk 2]
```diff
@@ -287,7 +288,7 @@ private boolean hasConstraintForDomain(LearnedConstraint lc, Class<? extends Con
         // but this is necessary so path is returned in #addEdge and stays in the worklist in #run
         return ImmutableSet.of(ImmutableList.of());
       }
-
+      Preconditions.checkState(isMethodInvocationNode(edge.parent), "Yield present on edge where parent is not MIT");
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java]: I am not really sure this is the place where we actually want to check this and I don't really see what's the risk you address with this invariant.

File: java-frontend/src/main/java/org/sonar/java/se/checks/ConditionAlwaysTrueOrFalseCheck.java (hunk 1 above)
Comment (saberduck) [inline java-frontend/src/main/java/org/sonar/java/se/checks/ConditionAlwaysTrueOrFalseCheck.java]: reverted the change

File: java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java (hunk 2 above)
Comment (saberduck) [inline java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java]: invariant check moved to `ExplodedGraph.Node.addParent`

**Pull request merged**